rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/TuplespaceAlg.scala

```diff
@@ -0,0 +1,81 @@
+package coop.rchain.rholang.interpreter.storage
+
+import cats.Parallel
+import cats.effect.Sync
+import coop.rchain.models.Channel.ChannelInstance.Quote
+import coop.rchain.models._
+import coop.rchain.models.TaggedContinuation.TaggedCont.ParBody
+import coop.rchain.rholang.interpreter.Dispatch
+import coop.rchain.rholang.interpreter.errors.ReduceError
+import coop.rchain.rspace.pure.PureRSpace
+import cats.implicits._
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.rholang.interpreter.storage.implicits._
+
+trait TuplespaceAlg[F[_]] {
+  def produce(chan: Channel, data: ListChannelWithRandom, persistent: Boolean): F[Unit]
+  def consume(binds: Seq[(BindPattern, Quote)], body: ParWithRandom, persistent: Boolean): F[Unit]
+}
+
+object TuplespaceAlg {
+  def rspaceTuplespace[F[_], M[_]](
+      pureRSpace: PureRSpace[F,
+                             Channel,
+                             BindPattern,
+                             ListChannelWithRandom,
+                             ListChannelWithRandom,
+                             TaggedContinuation],
+      dispatcher: => Dispatch[F, ListChannelWithRandom, TaggedContinuation])(
+      implicit F: Sync[F],
```

Comment:
Doing a quick search, in both cats and scalaz I can see that they don't really have a convention for this:
https://github.com/typelevel/cats/search?p=3&q=Monoid&unscoped_q=Monoid
https://github.com/scalaz/scalaz/search?q=Monoid&unscoped_q=Monoid

Still it would be good to have a standard for naming typeclass instances, especially when you have more than one typeclass instances